src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -325,7 +329,10 @@ private String validateScriptLanguage(String scriptLang) {
         return scriptLang;
     }
 
-    private String getScriptFromIndex(Client client, String scriptLang, String id) {
+    String getScriptFromIndex(String scriptLang, String id) {
+        if (client == null) {
+            throw new ElasticsearchIllegalArgumentException("Got an indexed script with no Client registered.");
```

Comment:
grrrr I didn't change this.... it's always been this way and to be honest I don't fully follow this whole optional client thing, and throwing an exception only from getScriptFromIndex but not from the other methods that use the client.... can we maybe clean this up in a separate PR?